app/uk/gov/hmrc/crsfatcaregistration/controllers/RegistrationWithUtrController.scala

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.crsfatcaregistration.controllers
+
+import com.google.inject.Inject
+import play.api.Logger
+import play.api.libs.json.{JsResult, JsSuccess, JsValue, Json}
+import play.api.mvc.{Action, ControllerComponents, Result}
+import uk.gov.hmrc.crsfatcaregistration.auth.AdminOnlyAuthAction
+import uk.gov.hmrc.crsfatcaregistration.config.AppConfig
+import uk.gov.hmrc.crsfatcaregistration.connectors.RegistrationWithUtrConnector
+import uk.gov.hmrc.crsfatcaregistration.models._
+import uk.gov.hmrc.http.HttpResponse
+import uk.gov.hmrc.play.bootstrap.backend.controller.BackendController
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Success, Try}
+
+class RegistrationWithUtrController @Inject() (
+  val config: AppConfig,
+  authenticate: AdminOnlyAuthAction,
+  registrationConnector: RegistrationWithUtrConnector,
+  override val controllerComponents: ControllerComponents
+)(implicit executionContext: ExecutionContext)
+    extends BackendController(controllerComponents) {
+
+  private val logger: Logger = Logger(this.getClass)
+
+  def sendAndRetrieveRegWithUtr: Action[JsValue] = authenticate(parse.json).async {
```

Comment:
I'd be inclined to include this functionality in the RegistrationController, keeping all the registration code in the same place.

You could then hook into the existing `withIdRegistration` function e.g.:

```
  def withUTROnly: Action[JsValue] = authenticate(parse.json).async {
    implicit request =>
      request.body.validate[UniqueTaxpayerReference].fold(
        invalid = _ => Future.successful(BadRequest("")),
        valid = utr => {
          withIdRegistration(buildPayload(utr))
        }
      )
  }
```